test/functional/legacy/autocmd_option_spec.lua

```diff
@@ -0,0 +1,120 @@
+        eq('', get_result())
+      end
+
+      before_each(function()
+        clear()
+
+        source([[
+          fu! AutoCommand(match)
+            let g:ret.=printf('Autocmd Option: <%s>,', a:match)
+            let g:ret.=printf(' OldVal: <%s>,', v:option_old)
+            let g:ret.=printf(' NewVal: <%s>,', v:option_new)
+            let g:ret.=printf(' Scope: <%s>', v:option_type)
+          endfu
+          
+          au OptionSet * :call AutoCommand(expand("<amatch>"))
+        ]])
+
+        execute([[let g:ret = '']])
+      end)
+
+      it('should set number option', function()
+        execute('set nu')
+        expected_combination('number', 0, 1, 'global')
+
+        execute('setlocal nonu')
+        expected_combination('number', 1, 0, 'local')
+
+        execute('setglobal nonu')
+        expected_combination('number', 1, 0, 'global')
+      end)
+
+      it('should set autoindent option',function()
+        execute('setlocal ai')
+        expected_combination('autoindent', 0, 1, 'local')
+
+        execute('setglobal ai')
+        expected_combination('autoindent', 0, 1, 'global')
+
+        execute('set noai')
+        expected_combination('autoindent', 1, 0, 'global')
+      end)
+
+      it('should invert global autoindent option',function()
+        execute('set ai!')
+        expected_combination('autoindent', 0, 1, 'global')
+      end)
+
+      it('should set several global list and number option',function()
+        execute('set list nu')
+        eq(expected_str('list', 0, 1, 'global') .. expected_str('number', 0, 1, 'global'),
+          get_result())
+      end)
+
+      it('should not print anything, use :noa.', function()
+        execute('noa set nolist nonu')
+        expected_empty()
+      end)
+
+      it('should set global acd', function()
+        execute('setlocal acd')
+        expected_combination('autochdir', 0, 1, 'local')
+      end)
+
+      it('should set autoread', function()
+        execute('set noar')
+        expected_combination('autoread', 1, 0, 'global')
+
+        execute('setlocal ar')
+        expected_combination('autoread', 0, 1, 'local')
+      end)
+
+      it('should invert global autoread', function()
+        execute('setglobal invar')
+        expected_combination('autoread', 1, 0, 'global')
+      end)
+
+      it('should set option backspace through :let', function()
+        execute('let &bs=""')
+        expected_combination('backspace', 'indent,eol,start', '', 'global')
+      end)
+
+      describe('setting option through setbufvar()', function()
+        it('shouldn\'t trigger because option name is invalid', function()
+          execute('call setbufvar(1, "&l:bk", 1)')
+          expected_empty()
+        end)
+
+        it('should trigger, use correct option name.', function()
+          execute('call setbufvar(1, "&backup", 1)')
+          expected_combination('backup', 0, 1, 'local')
+        end)
+      end)
+    end)
+  end)
+end)
```

Comment:
Apparently, I have mistaken what `bufmeths` means. It is good that my suggestion have found a bug before merge.